Unexpected broken link when running `linkcheck`

**Describe the bug**

I am including this link in my documentation:

https://wiki.stm32duino.com/index.php?title=Blue_Pill

Note that:

- I can visit that page with my browser
- I can `curl` that page as well from the terminal

However, when I run `make linkcheck` on my documentation, I get a broken-link report with the following error message: "406 Client Error: Not Acceptable".

Why am I getting that error? Is it possible to avoid it without having to use `linkcheck_ignore`?

**To Reproduce**

From a terminal (Bash):

```bash
python3 -m venv sphinx406
source sphinx406/bin/activate
pip install --upgrade pip
pip install sphinx
sphinx-quickstart --makefile --no-batchfile --sep -p Foo -a Bar -v 1.0.0 -r 1.0.0 -l en
echo '`Link <https://wiki.stm32duino.com/index.php?title=Blue_Pill>`_' > source/index.rst
make linkcheck SPHINXOPTS="-W"
```

**Expected behavior**

No error/warning displayed.

**Actual behavior**

An error/warning is displayed:

```
Running Sphinx v2.0.1
loading translations [en]... done
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [linkcheck]: targets for 1 source files that are out of date
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
preparing documents... done
writing output... [100%] index                                                                                                                        
(line    1) 

Warning, treated as error:
/tmp/asdf/source/index.rst.rst:1:broken link: https://wiki.stm32duino.com/index.php?title=Blue_Pill (406 Client Error: Not Acceptable for url: https://wiki.stm32duino.com/index.php?title=Blue_Pill)
make: *** [Makefile:19: linkcheck] Error 2
```

**Environment info**

- OS: Linux - Fedora 29
- Python version: 3.7.3
- Sphinx version: 2.0.1

--- a/CHANGES
+++ b/CHANGES
@@ -116,6 +116,7 @@ Bugs fixed
 * #6165: autodoc: ``tab_width`` setting of docutils has been ignored
 * #6311: autosummary: autosummary table gets confused by complex type hints
 * Generated Makefiles lack a final EOL (refs: #6232)
+* #6378: linkcheck: Send commonly used User-Agent
 
 Testing
 --------

--- a/sphinx/builders/linkcheck.py
+++ b/sphinx/builders/linkcheck.py
@@ -108,7 +108,8 @@ class CheckExternalLinksBuilder(Builder):
         kwargs = {
             'allow_redirects': True,
             'headers': {
-                'Accept': 'text/html,application/xhtml+xml;q=0.9,*/*;q=0.8'
+                'Accept': 'text/html,application/xhtml+xml;q=0.9,*/*;q=0.8',
+                'User-Agent': requests.useragent_header[0][1],
             },
         }
         if self.app.config.linkcheck_timeout:

